web/src/components/organisms/Asset/Asset/hooks.ts

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { Asset } from "@reearth-cms/components/molecules/Asset/asset.type";
+import { viewerRef } from "@reearth-cms/components/molecules/Asset/Asset/AssetBody/index";
+import {
+  PreviewType,
+  useGetAssetQuery,
+  useUpdateAssetMutation,
+} from "@reearth-cms/gql/graphql-client-api";
+
+export default (assetId?: string) => {
+  const [asset, setAsset] = useState<Asset>({} as Asset);
+  const [selectedPreviewType, setSelectedPreviewType] = useState<PreviewType>(PreviewType.Image);
+  useState<boolean>(false);
+  const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
+
+  const { data, loading } = useGetAssetQuery({
+    variables: {
+      assetId: assetId ?? "",
+    },
+  });
+
+  const [updateAssetMutation] = useUpdateAssetMutation();
+  const updateAsset = useCallback(
+    (assetId: string, previewType?: PreviewType) =>
+      (async () => {
+        if (!assetId) return;
+        const result = await updateAssetMutation({
+          variables: { id: assetId, previewType },
+          refetchQueries: ["GetAsset"],
+        });
+        if (result.errors || !result.data?.updateAsset) {
+          // TODO: notification
+          alert("Failed to update asset.");
+        }
+        if (result) {
+          // TODO: notification
+          alert("Asset was successfully updated.");
+        }
+      })(),
+    [updateAssetMutation],
+  );
+
+  useEffect(() => {
+    const asset: Asset = data?.asset as Asset;
+    setAsset(asset);
```

Comment:
```suggestion
    setAsset((data?.asset ?? {}) as Asset );
```
The syntax might be a little off, so please test this

Reply:
Done!
I tested it and it worked.